lib/internal/modules/esm/public_loader_proxy.js

```diff
@@ -0,0 +1,42 @@
+const { Worker } = require('internal/worker');
+
+
+/**
+ * The communication vehical constituting all memory shared between the main and the worker
+ */
+const comsChannel = new SharedArrayBuffer(2048);
+/**
+ * The lock/unlock segment of the shared memory. Atomics require this to be a Int32Array. This
+ * segment is used to tell the main to sleep when the worker is worker, and vice verse (for the
+ * worker to sleep whilst the main is working).
+ */
+const lock = new Int32Array(comsChannel, 0, 4);
+/**
+ * The request & response segment of the shared memory. TextEncoder/Decoder (needed to convert
+ * requests & responses into a format supported by the coms channel) reads and writes with
+ * Uint8Array.
+ */
+const requestResponseData = new Uint8Array(comsChannel, 4, 2044);
+
+const worker = new Worker('./worker.mjs', {
```

Comment:
A filename of `worker.mjs` implies that this is the only worker we’ll ever have under `modules/esm`. Perhaps `user_loaders_worker.mjs`? Also the actual file is named `worker.js`, a CommonJS file.

Reply:
I don't understand: it is a CJS file.

Reply:
The code says `worker.mjs`. If the file is named `worker.js`, shouldn’t the code say the same?

Re cross that bridge: renaming a file often messes with git history. It’s better to try to future-proof names as much as we can from the start. It doesn’t hurt to be as specific as possible with the filename, that will encourage future additions to be added into new files rather than bloating existing files.

Reply:
OH! I think it's copy+pasta from my experiment (which was esm).

One of the major downsides of using this worker is errors get swallowed.